tez-mapreduce/src/test/java/org/apache/tez/mapreduce/hadoop/TestMRInputHelpers.java

```diff
@@ -232,30 +278,32 @@ private DataSourceDescriptor generateDataSourceDescriptorMapRed(Path inputSplits
   @Test(timeout = 5000)
   public void testInputSplitLocalResourceCreationWithDifferentFS() throws Exception {
     FileSystem localFs = FileSystem.getLocal(conf);
-    Path LOCAL_TEST_ROOT_DIR = new Path("target"
-        + Path.SEPARATOR + TestMRHelpers.class.getName() + "-localtmpDir");
+    Path splitsDir = localFs.resolvePath(LOCAL_TEST_ROOT_DIR);
 
-    try {
-      localFs.mkdirs(LOCAL_TEST_ROOT_DIR);
-
-      Path splitsDir = localFs.resolvePath(LOCAL_TEST_ROOT_DIR);
+    DataSourceDescriptor dataSource = generateDataSourceDescriptorMapRed(splitsDir);
 
-      DataSourceDescriptor dataSource = generateDataSourceDescriptorMapRed(splitsDir);
-
-      Map<String, LocalResource> localResources = dataSource.getAdditionalLocalFiles();
+    Map<String, LocalResource> localResources = dataSource.getAdditionalLocalFiles();
 
-      Assert.assertEquals(2, localResources.size());
-      Assert.assertTrue(localResources.containsKey(
-          MRInputHelpers.JOB_SPLIT_RESOURCE_NAME));
-      Assert.assertTrue(localResources.containsKey(
-          MRInputHelpers.JOB_SPLIT_METAINFO_RESOURCE_NAME));
+    Assert.assertEquals(2, localResources.size());
+    Assert.assertTrue(localResources.containsKey(
+        MRInputHelpers.JOB_SPLIT_RESOURCE_NAME));
+    Assert.assertTrue(localResources.containsKey(
+        MRInputHelpers.JOB_SPLIT_METAINFO_RESOURCE_NAME));
 
-      for (LocalResource lr : localResources.values()) {
-        Assert.assertFalse(lr.getResource().getScheme().contains(remoteFs.getScheme()));
-      }
-    } finally {
-      localFs.delete(LOCAL_TEST_ROOT_DIR, true);
+    for (LocalResource lr : localResources.values()) {
+      Assert.assertFalse(lr.getResource().getScheme().contains(remoteFs.getScheme()));
     }
   }
 
+  @Before
+  public void before() throws IOException {
+    FileSystem localFs = FileSystem.getLocal(conf);
```

Comment:
We are getting the `localFs` multiple times, can we store it as a class variable just like the `remoteFs`

Reply:
ack